parallax/probe_calibration.py

```diff
@@ -3,125 +3,296 @@
 - local space: Stage coordinates
 - global space: Reticle coordinates
 """
-from PyQt5.QtCore import QObject
+from PyQt5.QtCore import QObject, pyqtSignal
+from sklearn.linear_model import LinearRegression
 import logging
 import numpy as np
-import cv2
+import pandas as pd
 import csv
+import os
 
 # Set logger name
 logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
 # Set the logging level for PyQt5.uic.uiparser/properties to WARNING, to ignore DEBUG messages
 logging.getLogger("PyQt5.uic.uiparser").setLevel(logging.WARNING)
 logging.getLogger("PyQt5.uic.properties").setLevel(logging.WARNING)
 
 class ProbeCalibration(QObject):
+    """
+    Handles the transformation of probe coordinates from local (stage) to global (reticle) space.
+
+    Attributes:
+        calib_complete_x (pyqtSignal): Signal emitted when calibration in the X direction is complete.
+        calib_complete_y (pyqtSignal): Signal emitted when calibration in the Y direction is complete.
+        calib_complete_z (pyqtSignal): Signal emitted when calibration in the Z direction is complete.
+        calib_complete (pyqtSignal): General signal emitted when calibration is fully complete.
+
+    Args:
+        stage_listener (QObject): The stage listener object that emits signals related to stage movements.
+    """
+    calib_complete_x = pyqtSignal()
+    calib_complete_y = pyqtSignal()
+    calib_complete_z = pyqtSignal()
+    calib_complete = pyqtSignal(str, object)
+    transM_info = pyqtSignal(object, float, object)
+
     """Class for probe calibration."""
     def __init__(self, stage_listener):
-        """ Initialize the Probe Calibration object. """
+        """
+        Initializes the ProbeCalibration object with a given stage listener.
+        """
         super().__init__()
         self.stage_listener = stage_listener
-        self.stage_listener.probeCalibRequest.connect(self.local_global_transform)
+        self.stage_listener.probeCalibRequest.connect(self.update)
         self.stages = {}
-        self.local_points = []
-        self.global_points = []
+        self.df = None
         self.inliers = []
-        self.transform_matrix = None
-        self.error_min = 1000
-        
+        self.stage = None
+        self.threshold_min_mix = 2000 #TODO 
```

Comment:
```suggestion
        self.threshold_min_max = 2000 #TODO 
```

Reply:
Fixed! I guess I have a dyslexia! It took few minutes to see what is difference.